doc/user/content/overview/arrangements.md

```diff
@@ -0,0 +1,130 @@
+
+## Dataflows
+
+Materialize can make incremental updates efficiently because it's built on an incremental data-parallel compute engine, [Differential Dataflow](https://timelydataflow.github.io/differential-dataflow/introduction.html), which in turn is built on a distributed processing framework called [Timely Dataflow](https://timelydataflow.github.io/timely-dataflow/).
+
+When you create a materialized view and issue a query, Materialize creates a **dataflow**. A dataflow consists of instructions on how to respond to data input. Once executed, the dataflow computes and stores the result of the SQL query in memory, polls the source for updates, and then incrementally updates the query results when new data arrives.
+
+### Collections
+
+Materialize dataflows act on **collections** of data, [multisets](https://en.wikipedia.org/wiki/Multiset) that store each event in an update stream as a triple of `(data, time, diff)`.
+
+Term | Definition
+-----|-----------
+**data**  |  The record update.
+**time**  |  The logical timestamp of the update.
+**diff**  |  Number of copies of the record (`-1` for deletion, `1` for addition).
+
+## Arrangements
+
+To transform the source input into the dataflow into the result output, operators (independent computational components)
+build and maintain indexes on the input and output collections so that they can quickly access collection data in the future. Because queries can overlap, different operators often need to build the exact same indexes for multiple queries. Instead of asking operators to perform redundant work, Materialize builds the index once and maintains it in memory, sharing the required resources across all the operators that use the indexed data. The index is then effectively a sunk cost, and the cost of each query is determined only by the new work it introduces.
+
+We call the indexed update stream an **arrangement**.
+
+In addition to materialized views, certain internal operators, like `reduce` and `join`, also create arrangements so the operators can respond efficiently to updates.
+
+### Arrangement structure
+
+In the `(data, time, diff)` triples that make up the update stream, `data` is structured as a key-value pair.  Arrangements are indexed on the `data` keys. You cannot manually create an arrangement, although you can influence the shape of the arrangements created by Materialize by the way you structure queries and views.
+
+When creating an arrangement for a join where the user has not specified a key, Materialize attempts to choose a key that will ensure that data is well distributed. If there is a primary key, that will be used; if there are source fields not required by the query, they are not included. Often Materialize can pull primary key info from a Confluent schema.
```

Comment:
I don't think the first sentence is correct. When the user causes a join to exist, the join either has a key or does not have a key, and we don't change that. When a user creates an arrangement *that is ambivalent about its key* MZ will choose a well-distributed key. I think this mostly only even happens with `CREATE MATERIALIZED ...` and perhaps `CREATE DEFAULT INDEX ...`, but every case MZ creates an arrangement otherwise, it is for a specific use and with a specific key in mind.